src/Microsoft.OData.Core/SelectedPropertiesNode.cs

```diff
@@ -312,8 +321,7 @@ internal SelectedPropertiesNode GetSelectedPropertiesForNavigationProperty(IEdmS
 
                 // try to find a child with a type segment before it that matches the current type.
                 // Note: the result of this aggregation will be either empty or a found child node.
-                return this.GetMatchingTypeSegments(structuredType)
-                    .Select(typeSegmentChild => typeSegmentChild.GetSelectedPropertiesForNavigationProperty(structuredType, navigationPropertyName))
+                return GetSelectePropertiesForTypeSegmentsNavigationProperties(this.GetMatchingTypeSegments(structuredType), structuredType, navigationPropertyName)
                     .Aggregate(child, CombineNodes);

```

Comment:
Use lambda to save the allocation?

 (l, r) => CombineNodes(l, r)

Reply:
Done